### ISSUE TITLE
Suggestion to make the example easier (Scanner not visible)

### ISSUE THREAD

**Issue by MariusBozem** [ideacreation/react-native-barcodescanner#11]
Hey, I just tried our the barcode scanner. At first I couldn't get it to work and was confused what my mistake was.

It turned out that the container <View> element did not have "flex: 1" in its styles and therefore apparently had a height of 0 which meant that the barcode scanner wasn't visible at all. I don't know if others will stumble upon this aswell, but just in case, here's how I solved it:

in render(): 
`<View style={styles.container}>
        <BarcodeScanner
          onBarCodeRead={this.barcodeReceived}
          style={{ flex: 1 }}
          torchMode={this.state.torchMode}
          cameraType={this.state.cameraType}
          />
      </View>`
and the styles:

`
var styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'red',
  }
});
`

**Comment by MariusBozem** [comment]
Also the scanner seems to always be in landscape mode regardless of the device orientation. Is there a fix for that?

**Comment by andreyvital** [comment]
@MariusBozem I think that it isn't responsive to orientation changes. I'm willing to code this specific feature if necessary. But before we get to that, we have to discuss the API. 

For example: it may be interesting to others a prop in `BarcodeScanner` component which allows you to set the orientation and then you can handle it in your application level &ndash; or &ndash; you can still have a prop that allows you to set the orientation and if isn't set then it starts responding to orientation changes automatically.

**Comment by MariusBozem** [comment]
Yes, I like both of your suggestions and for me it would be perfect if it would respond to orientation automatically for now

**Comment by andreyvital** [comment]
By looking at the `dm77/barcodescanner` (the library that this package uses for scanning) looks like it's already there: [ViewFinderView.java#L149-L158](https://github.com/dm77/barcodescanner/blob/5efbf424d8012ca53c0d1cdcdff14e25ae40a010/core/src/main/java/me/dm7/barcodescanner/core/ViewFinderView.java#L149-L158)

**Comment by MariusBozem** [comment]
Yes,  I've done a bit of research and can confirm that the used library should support it.

I've only tried it on an emulator so far and it did not respond to orientation changes though

**Comment by andreyvital** [comment]
Well, I fixed the issue #9 which is about `viewFinderDrawLaser`, so I'm going to look about handling the orientation changes now and then submit a PR for both.

/cc @andreaskeller

**Comment by andreyvital** [comment]
@MariusBozem 

I tried to debug but we're out of luck. It seems that `DisplayUtils.getScreenOrientation(context)` always returns `1` which is `Configuration.PORTRAIT`. I also tried to get current device orientation instead of application orientation &ndash; but all methods I've tried returned `1`.

I don't have sufficient knowledge to debug what's going on :(. Maybe it's something related to RN rather than the library itself. Do you have any ideas of what can we do to solve this?

I also think that my first thought about introducing a new property which would allows us to define the initial orientation falls through, as `ZXingScannerView` doesn't provide an API to change it explicitly neither allows to change `mFramingRect` &mdash; the library handles orientation internally.

:sob:

**Comment by MariusBozem** [comment]
First of all, thanks for looking into it this quickly. Unfortunately I have to work on other parts of the app for now so I can't look into it right now, but I'll try to do it once I have finished the other parts.

**Comment by andreaskeller** [comment]
©CentaurWarchief thank you for your help. I will look into the orientation change.

**Comment by MariusBozem** [comment]
@andreaskeller Do you have an update on the orientation issue? Unless I'm doing sth. wrong, it doesn't work with portrait mode because the scanner is rendered as though it was landscape mode.

**Comment by ddzyndra** [comment]
Did anybody solved this issue with orientation ?
